File: mergeTwoSortedArrays.java
import java.io.*;
import java.util.* ;

public class Solution {
    public static int[] ninjaAndSortedArrays(int arr1[], int arr2[], int m, int n) {
        // Write your code here.
        int[] ans = new int[n+m];
        int k = 0;
        int i = 0; 
        int j = 0;
        while(j<n && i<m) {
            if(arr1[i] < arr2[j]) {
                ans[k++] = arr1[i];
                i++;
            } else {
                ans[k++] = arr2[j];
                j++;
            }
        }
        while(i < m) {
            ans[k++] = arr1[i];
            i++;
        }
        while(j < n) {
            ans[k++] = arr2[j];
            j++;
        }
        return ans;
    }
}

File: MergeTwoSortedArrays.cpp
vector < int > sortedArray(vector < int > a, vector < int > b) {
    // Write your code here
    int i = 0, j = 0;
    vector<int> sortedA;
    while(i < a.size() && j < b.size()) {
        if(a[i] < b[j]) {
            auto it = find(sortedA.begin(), sortedA.end(), a[i]);
            if(it == sortedA.end())
                sortedA.push_back(a[i]);
            i++;
        } else {
            auto it = find(sortedA.begin(), sortedA.end(), b[j]);
            if(it == sortedA.end())
                sortedA.push_back(b[j]);
            j++;
        }
    }

    while(i < a.size()) {
        auto it = find(sortedA.begin(), sortedA.end(), a[i]);
        if(it == sortedA.end())
            sortedA.push_back(a[i]);
        i++;
    }

    while(j < b.size()) {
        auto it = find(sortedA.begin(), sortedA.end(), b[j]);
        if(it == sortedA.end())
            sortedA.push_back(b[j]);
        j++;
    }

    return sortedA;
}